src/python/turicreate/toolkits/activity_classifier/_tf_model_architecture.py

```diff
@@ -112,46 +126,75 @@ def __init__(self, net_params, batch_size, num_features, num_classes, prediction
+                    self.sess.run(_tf.assign(_tf.get_default_graph().get_tensor_by_name(key+":0"), net_params[key] ))
             if key in biases.keys():
-                self.sess.run(_tf.assign(_tf.get_default_graph().get_tensor_by_name(key+":0"), net_params[key].asnumpy()))
+                self.sess.run(_tf.assign(_tf.get_default_graph().get_tensor_by_name(key+":0"), net_params[key]))
 
 
     def train(self, feed_dict):
 
-        _, loss, acc, probs = self.sess.run([self.train_op, self.loss_op, self.acc, self.probs], 
+        for key in feed_dict.keys():
+            feed_dict[key] = _np.array(feed_dict[key], copy=False)
+            feed_dict[key] =  _np.reshape(feed_dict[key], (feed_dict[key].shape[0], feed_dict[key].shape[2], feed_dict[key].shape[3]))
+        _, loss, probs = self.sess.run([self.train_op, self.loss_op, self.probs], 
             feed_dict={self.data : feed_dict['input'], self.target : feed_dict['labels'], self.weight : feed_dict['weights'], self.is_training : True})
-        result = {'loss' : loss, 'acc': acc, 'probs': probs }
+        
+        prob = _np.array(probs)
+        probabilities = _np.reshape(prob, (prob.shape[0], prob.shape[1]*prob.shape[2]))
+        result = {'loss' : _np.array(loss), 'output': probabilities }
+
         return result
 
 
     def predict(self, feed_dict):
 
-        loss, acc, probs= self.sess.run([self.loss_op, self.acc, self.probs], 
-            feed_dict={self.data : feed_dict['input'], self.target : feed_dict['labels'], self.weight : feed_dict['weights'], self.is_training: False})
-        result = {'loss' : loss, 'acc': acc, 'probs': probs }
+
+
+        for key in feed_dict.keys():
+            feed_dict[key] = _np.array(feed_dict[key], copy=False)
+            feed_dict[key] =  _np.reshape(feed_dict[key], (feed_dict[key].shape[0], feed_dict[key].shape[2], feed_dict[key].shape[3]))
+
+        if len(feed_dict.keys()) == 1 :
+            probs = self.sess.run(self.probs, 
+                feed_dict={self.data : feed_dict['input'], self.is_training: False})
+            prob = _np.array(probs)
+            probabilities = _np.reshape(prob, (prob.shape[0], prob.shape[1]*prob.shape[2]))
+            result  = { 'output' :  probabilities}
+            
+        else:
+            loss,  probs= self.sess.run([self.loss_op,  self.probs], 
+                feed_dict={self.data : feed_dict['input'], self.target : feed_dict['labels'], self.weight : feed_dict['weights'], self.is_training: False})
+            prob = _np.array(probs)
+            probabilities = _np.reshape(prob, (prob.shape[0], prob.shape[1]*prob.shape[2]))
+            result = {'loss' : _np.array(loss),  'output': probabilities }
+        
         return result
 
 
     def export_weights(self):
 
-        # Export weights in Core ML Spec
+        # Export weights in Core ML Spec 
```

Comment:
Please refrain from adding trailing whitespace. Does clang-format or other tool work on Python code?

Reply:
I don't think it works on python code